app/src/components/connect/SessionRow.tsx

```diff
@@ -119,29 +133,54 @@ const SessionRow: React.FC<Props> = ({ session, style }) => {
     sessionStore.revokeSession(session);
   }, [session]);
 
+  const toggleQRModal = useCallback(() => setShowQR(v => !v), []);
+
   const { Row, Column, ActionColumn, CloseIcon } = Styled;
   return (
     <Row style={style}>
       <ActionColumn>
         {session.isPaired ? (
-          <Tip overlay={l('paired')}>
-            <Copy disabled />
-          </Tip>
+          <>
+            <Tip overlay={l('paired')}>
+              <BoltOutlined disabled />
+            </Tip>
+            <Tip overlay={l('paired')}>
+              <Copy disabled />
+            </Tip>
+            <Tip overlay={l('paired')}>
+              <QRCode disabled />
+            </Tip>
+          </>
         ) : (
-          <Tip overlay={l('copy')}>
-            <Copy onClick={handleCopy} />
-          </Tip>
+          <>
+            <Tip overlay={l('pairTerminal')}>
+              <a href={session.terminalConnectUrl} target="_blank" rel="noreferrer">
+                <BoltOutlined />
+              </a>
+            </Tip>
+            <Tip overlay={l('copy')}>
+              <Copy onClick={handleCopy} />
+            </Tip>
+            <Tip overlay={l('generateQR')}>
+              <QRCode onClick={toggleQRModal} />
+            </Tip>
+          </>
         )}
+        <QRCodeModal
+          url={session.terminalConnectUrl}
```

Comment:
What if a user wants to pair to a mobile app other than Terminal?

Reply:
Good question. The mobile app dev can parse the Terminal url to extract the encoded phrase & server, then decode it.

If we used a different format for the QR data, it would prevent the user from scanning the QR with any scanning app on their device and being directed to Terminal in the browser. From a UX perspective, I think most users immediately open their phone's camera app when they see a QR code in the wild. With a different format, they'd have to know to go to the Terminal website first, then click on a "Scan QR" button. This would have to be explicitly spelled out for them in the modal. We just eliminated the extra step.

Do you think this prohibits other apps from using the QR codes? Do you have any suggestions on how to improve this?

Reply:
The option to use the pairing phrase is still available, correct? If a user wants to connect a different app they could simply type the pairing phrase. Not as user friendly but it means users aren't totally locked out from using 3rd party apps

Reply:
Mobile apps would be able to parse it, just have to decode it.

I'll have to put some thought into how we could do this better.